gradle/lib/dependabot/gradle/file_updater.rb

```diff
@@ -150,9 +151,25 @@ def original_buildfile_declarations(dependency, requirement)
             next false unless line.match?(name_regex)
           end
 
-          line.include?(requirement.fetch(:requirement))
+          if dependency.name.include?(":")
+            group, name = dependency.name.split(":")
+            version = requirement.fetch(:requirement)
+
+            pattern = if line.include?("group")
+                        # rubocop:disable Layout/LineLength
+                        /group\s*[=:]\s*['"]#{group}['"]\s*,\s*name\s*[=:]\s*['"]#{name}['"]\s*,\s*version\s*[=:]\s*['"]#{version}['"]/
+                        # rubocop:enable Layout/LineLength
+                      else
```

Comment:
This regex logic is very brittle, it took me many iterations to accommodate all the different specs that we have

Is there any better way to do this? (without native helpers?)

Sadly, there are a lot of different ways to declare a dependency in gradle and that does not help

Reply:
Also, I decided to disable `Layout/LineLength` because I think that multiple lines would make the regex even more confusing. But we can revert it if needed

Reply:
Agreed that this isn't ideal - Gradle does support other styles of declaring dependencies in its DSL including a [map notation](https://github.com/gradle/gradle/blob/master/subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/DependencyHandler.java#L159) and it isn't clear if/how it's supported here

Renovate has a [Gradle parser](https://github.com/renovatebot/renovate/pull/75210) that we could shell out to but I think we can open it as a separate issue since it's pretty involved

Reply:
I agree. At some point, I think that we should see if we can leverage native Gradle code directly to do all of this